src/blueprints/tv2/action-factories/tv2-audio-action-factory.ts

```diff
@@ -225,7 +226,7 @@ export class Tv2AudioActionFactory {
     const audioAction: Tv2FadeAudioBedAction = action as Tv2FadeAudioBedAction
 
     const fadeDurationInMilliseconds: number =  this.getTimeFromFrames(this.isInteger(fadeDurationInFrames) ? fadeDurationInFrames : audioAction.metadata.defaultFadeDurationInFrames)
-    const fadeAudioBedTimelineObject: TimelineObject = this.casparCgTimelineObjectFactory.createFadeAudioBedTimelineObject(fadeDurationInMilliseconds)
+    const fadeAudioBedTimelineObject: Tv2BlueprintTimelineObject = this.videoClipTimelineObjectFactory.createFadeAudioBedTimelineObject(fadeDurationInMilliseconds)
```

Comment:
The `.createFadeAudioBedTimelineObject` sounds like a method that should be on a `audioTimelineObjectFactory`, or am i misunderstanding what a 'AudioBed' is?

Perhabs it is just the name of the method that is incorrect.

Reply:
I understand your confusion. I had a hard time finding a place to put that method.
The trick is, it's CasparCG that plays the audiobed and it's also CasparCG that fades the audiobed.
So we need some way to get a TimelineObjectFactory for CasparCG when fading the audiobed. 
Putting `createFadeAudioBedTimelineObject` on `Tv2AudioTimelineObjectFactory` and then create a `Tv2CasparCgAudioTimelineObjectFactory` would mean all but one of the methods on the interface would have to throw an unsupported exception. Also, how would we differentiate between which TimelineObjectFactory to use in the AudioActionFactory because we still need the `Tv2SisyfosAudioTimelineObjectFactory`?

I agree, the current implementation isn't optimal and I'm open for suggestions.

Reply:
That explanation helped. I see now that the current implementation is the better of 2 rotten tomatoes.
Don't have any better suggestion at the moment.